### PR TITLE
Diamond

### PR DESCRIPTION
**Pull request by RNAer**

### CODE REVIEW

File: recipes/Diamond/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  number: 0
+
+test:
+  commands:
+    - diamond -h
+
+about:
+  summary: a BLAST-compatible local aligner for mapping protein and translated DNA query sequences against a protein reference database
+  home: http://github.com/bbuchfink/diamond
+  license: Copyright (c) 2014, University of Tuebingen
```
Comment (Jorge-C) [inline recipes/Diamond/meta.yaml]: The license is actually new BSD (https://en.wikipedia.org/wiki/BSD_licenses#3-clause_license_.28.22Revised_BSD_License.22.2C_.22New_BSD_License.22.2C_or_.22Modified_BSD_License.22.29)

File: recipes/Diamond/build.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+cd src
+./install-boost &> /dev/null
```
Comment (Jorge-C) [inline recipes/Diamond/build.sh]: Does boost actually end up being packaged in the binary? From looking at that script, it doesn't seem to be copied to $PREFIX.

However, the cmake build instructions do seem to do that well, so maybe they should be used? https://github.com/bbuchfink/diamond#compiling-using-cmake

Comment (RNAer) [inline recipes/Diamond/build.sh]: hah, the compiling instruction is different in the readme. I'll try cmake

**Comment by RNAer** [comment]
the release version dont have cmake file in it. @Jorge-C , do you know if we can use commit hash for version and download? and does conda build env have cmake?

**Comment by Jorge-C** [comment]
I'm not sure about using commit hashes, but keeping just `git_url` should use latest master. The only change so far from 0.7.9 is the addition of cmake.

To use cmake on travis, we need to [add it](http://docs.travis-ci.com/user/apt/#Adding-APT-Packages) to `.travis.yml`.

**Comment by RNAer** [comment]
ok. I tried cmake locally. It doesn't make a difference. boost is still
required to install manually. The problem on linux is that:
https://travis-ci.org/biocore/conda-recipes/jobs/90202664#L374

**Comment by colinbrislawn** [comment]
There is a conda package for `boost` which may work well. You could try adding boost to the requirements then building like this:
https://github.com/bbuchfink/diamond#compiling-from-source

**Comment by RNAer** [comment]
it is not boost. boost is compiled well. it is the compilation of diamond
on linux itself.

File: recipes/Diamond/build.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+mkdir build
+cd build
+cmake .. # Use cmake -DCMAKE_INSTALL_PREFIX=... to install to a different prefix.
+make install
+mkdir -p $PREFIX/bin
+cp diamond $PREFIX/bin
```
Comment (Jorge-C) [inline recipes/Diamond/build.sh]: These two last lines are not needed if you use `DCMAKE_INSTALL_PREFIX` instead

File: recipes/Diamond/build.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+conda install boost
+
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$PREFIX/bin ..
```
Comment (Jorge-C) [inline recipes/Diamond/build.sh]: Won't solve the boost problems, but the prefix should be $PREFIX, not $PREFIX/bin

**Comment by patricksnape** [comment]
Hey guys! I think I have a working build of this:

**meta.yaml**

```
package:
  name: diamond
  version: "0.7.9.dev0"

build:
  number: 0

requirements:
  build:
   - cmake 3.3.0  # [not osx]
   - boost 1.59.0

  run:
   - boost 1.59.0

test:
  commands:
    - diamond -h

about:
  summary: a BLAST-compatible local aligner for mapping protein and translated DNA query sequences against a protein reference database
  home: http://github.com/bbuchfink/diamond
  license: Modified BSD License
  license_file: src/COPYING
```

**build.sh**

```
#!/bin/bash
cp -r $RECIPE_DIR/../ .

mkdir build
cd build

CMAKE_ARCH="-m"$ARCH
INCLUDE_PATH=${PREFIX}/include
LIBRARY_PATH=${PREFIX}/lib

export LDFLAGS="-L${LIBRARY_PATH} $LDFLAGS"

cmake -LAH .. \
-DCMAKE_PREFIX_PATH=$PREFIX \
-DCMAKE_INSTALL_PREFIX=$PREFIX \
-DBOOST_ROOT=$PREFIX \
-DBOOST_INCLUDEDIR="${INCLUDE_PATH}" \
-DBOOST_LIBRARYDIR="${LIBRARY_PATH}" \

cmake --build . --config Release --target install
```

There are a few things to note here though:
1. This relies on the [menpo/conda-boost](https://github.com/menpo/conda-boost) package to work. The easiest way to get this to build would be to add menpo to your default channels. Alternatively, you could copy the menpo recipe and build/host it yourselves.
2. This assumes that the conda folder is local the the rest of the source code (with is indicated by `cp -r $RECIPE_DIR/../ .` which copies the source into the build directory). I had to do this because the currently released diamond doesn't contain the `CMakeLists.txt`. Alternatively, you could make the source the master git repository which would have the same effect - but make versioning difficult! Notice that I appended a `dev0` to the version because I wasn't sure what the next version would be.
3. There is a bug in the `CMakelists.txt` at the moment. `pthread` is missing: `target_link_libraries(diamond blast_core ${Boost_LIBRARIES} -lpthread)`

Hope that helps!

**Comment by patricksnape** [comment]
In reference to your other [issue](https://github.com/bbuchfink/diamond/issues/23), this is because you need to set the `BOOST_ROOT` etc as I have done. This may work with the built in boost package but I haven't tried.

**Pull request merged**